src/gitutil.js

```diff
@@ -43,12 +43,14 @@ exports.tagExists = function*(tagName) {
 }
 
 exports.retrieveCurrentBranchName = function*(allowDetached) {
-    var ref = yield executil.execHelper(executil.ARGS('git symbolic-ref HEAD'), true, true);
-    if (!ref) {
+    var ref;
+    try {
+        ref = yield executil.execHelper(executil.ARGS('git symbolic-ref HEAD'), true, true);
```

Comment:
Fixed this - removed the use of `pretend` as it would not work across control flow anyway.